### ISSUE TITLE
Printing of inserted hidden lambdas

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3238]
The printing of inserted hidden lambdas has been irritating me for years.
And for newcomers it is very confusing.

Short test case:
```agda
open import Agda.Builtin.Equality
  
_∘_ : ∀ {a b c}
        {A : Set a} {B : A → Set b} {C : {x : A} → B x → Set c} →
        (f : ∀ {x} (y : B x) → C y) →
        (g : (x : A) → B x) →
        ((x : A) → C (g x))
f ∘ g = λ x → f (g x)

postulate
  A : Set
  B : A → Set
  C : {x : A} → B x → Set
  f : ∀ {x} (y : B x) → C y
  g : (x : A) → B x 

test : (f ∘ g) ≡ (f ∘ g)
test = {!!}

-- GOAL:
-- ?0 : ((λ {;x} → f) ∘ g) ≡ ((λ {;x} → f) ∘ g)

-- EXPECTED:
-- ?0 : (f ∘ g) ≡ (f ∘ g)
```

**Comment by JacquesCarette** [comment]
Emphatically agreed! I am no longer a newcomer, and this still takes me some time to parse properly.

**Comment by gallais** [comment]
So I had a look at this test case and it turns out that the recorded `Origin`
of the implicit `x` is `UserWritten` according to Agda. We need to fix that
bug first.

**Comment by gallais** [comment]
I cleaned up the `Doc` generated by `Concrete.Pretty` to remove hidden lambda
abstractions inserted by the machine.

However the translation from Abstract to Concrete seems to be pretty keen to
insert `Paren` constructors so I get the following:

```agda
((f) ∘ g) ≡ ((f) ∘ g)
```

I will have a crack at it after lunch.

**Comment by gallais** [comment]
The fix in #3944 naturally changed some of the error messages in the test suite.
If you are happy with the modifications (the most disruptive change is probably
that some sizes are now implicit in error messages), do not hesitate to merge.

**Comment by xekoukou** [comment]
Probably , this introduced the printing "error" in #4160 :
https://github.com/agda/agda/commit/eab488ef45a8017e2c8df71f11c3e992dc30f6e0#diff-36dccb1101d64196a959ddd699f4ef2cR709
